modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectFieldResourceImpl.java

```diff
@@ -266,6 +284,38 @@ private ObjectField _toObjectField(
 			objectField);
 	}
 
+	private void _validateAndCreateListTypeDefinition(ObjectField objectField)
+		throws Exception {
+
+		if (Objects.equals(
+				objectField.getBusinessTypeAsString(),
+				ObjectFieldConstants.BUSINESS_TYPE_PICKLIST)) {
+
+			if (Objects.isNull(
+					objectField.getListTypeDefinitionExternalReferenceCode())) {
+
+				throw new ObjectFieldListTypeDefinitionException(
+					StringBundler.concat(
+						"List type definition external reference code must ",
+						"not be null when business type is ",
+						ObjectFieldConstants.BUSINESS_TYPE_PICKLIST));
+			}
```

Comment:
@murilostodolni I would rather keep the logic as follows:

```
if (Objects.equals(
		objectField.getBusinessTypeAsString(),
		ObjectFieldConstants.BUSINESS_TYPE_PICKLIST) &&
	Validator.isNotNull(
		objectField.getListTypeDefinitionExternalReferenceCode()))
```

and don't throw any exceptions here for two reasons:

1. This code will not work for people who are already creating PickList fields and not sending the external reference code. This means that some Poshi tests will also fail.
3. We should not validate anything in the `ResourceImpl`. In this case, if the `listTypeDefinitionExternalReferenceCode` is `null`, we should send to the local service the value `0` for the `listTypeDefinitionId` (I believe we should change the `ObjectFieldUtil.getListTypeDefinitionId` for this) and then you can perform this validation in the `ObjectFieldLocalServiceImpl`.